### DISCUSSION TITLE
[Vue] Floating element with `v-if` not displayed

### DISCUSSION THREAD

**Discussion by bgrand-ch** [floating-ui/floating-ui#2569] [Q&A] [answered]
Hello @atomiks,

Thanks a lot for Floating ui!

Context:
- `"@floating-ui/vue": "1.0.2"`
- `"nuxt": "3.7.3"` with TypeScript
- `"quasar": "2.12.7"`
- `"grapesjs": "0.21.6"`

I'm having a problem on Vue with my floating element containing form fields.

I read this [paragraph](https://floating-ui.com/docs/vue#:~:text=Ensure%20you%20are%20using%20conditional%20rendering) in the doc and this [discussion](https://github.com/floating-ui/floating-ui/discussions/2274#discussion-5040517).

When I use `v-show`, my floating element is displayed, but fields inside can't have focus and I can't use them.

When I use `v-if`, my floating element is **not** displayed.

I've tried displaying my fields outside the floating element and they work fine.

Could you help me please?

---

```vue
<template>
<!-- ...more... -->

<!-- Not displayed -->
<div
  v-if="hasBrickStyle"
  ref="brickStyle"
  class="absolute z-max q-pa-md bg-white shadow-1 rounded-borders"
  style="width: 240px; min-height: 100px;"
  :style="floatingStyles"
>
  <div class="column q-gutter-y-sm">
    <q-input
      v-for="brickStyle in brickStyles"
      v-model="brickStyleProps[brickStyle.property!]"
      :label="brickStyle.label"
      :debounce="500"
      outlined
      dense
      @update:model-value="updateBrickStyle(brickStyle)"
    />
  </div>
</div>

<!-- Displayed -->
<!-- <div
  v-show="hasBrickStyle"
  ref="brickStyle"
  class="absolute z-max q-pa-md bg-white shadow-1 rounded-borders"
  style="width: 240px; min-height: 100px;"
  :style="floatingStyles"
>
  <div class="column q-gutter-y-sm">
    <q-input
      v-for="brickStyle in brickStyles"
      v-model="brickStyleProps[brickStyle.property!]"
      :label="brickStyle.label"
      :debounce="500"
      outlined
      dense
      @update:model-value="updateBrickStyle(brickStyle)"
    />
  </div>
</div> -->

<!-- ...more... -->
</tempate>

<script setup lang="ts">
// ...more...

import {
  useFloating,
  autoUpdate,
  autoPlacement,
  shift,
  offset
} from '@floating-ui/vue'
import brickStyles from 'data/brick-styles'

const brickStyleProps = reactive<any>({
  width: '',
  height: ''
})

const hasBrickStyle = ref(false)
const brickStyle = ref<HTMLDivElement|null>(null)
const selectedEl = ref<HTMLElement|null>(null) // element in GrapesJS iframe

const { floatingStyles } = useFloating(selectedEl, brickStyle, {
  whileElementsMounted: autoUpdate,
  placement: 'right',
  middleware: [
    offset(10),
    autoPlacement(),
    shift({ padding: 10 })
  ]
})

// ...more...
<script>
```

**Comment by atomiks** [comment]
Where is `selectedEl` ref in the template? Please try to provide a reproduction online to view

**Reply by bgrand-ch** [reply, marked as answer]
Thanks for quick answer.

The reproduction link:
https://codesandbox.io/p/sandbox/grapesjs-floating-ui-ys4tr6

But it works in this case...

I was able to understand that the error came from my code, thank you very much!
